R/dvhydrograph-render.R

```diff
@@ -3,199 +3,291 @@ dvhydrographPlot <- function(data) {
-    
-    plot_object <- gsplot(ylog = dvInfo$logAxis, yaxs = 'i') %>%
+  }
+
+  #Get Basic Plot data
+  dvData <- list()
+
+  dvData[['stat1TimeSeries']] <- parseDVTimeSeries(reportObject, 'firstDownChain', 'downChainDescriptions1', timezone, excludeZeroNegativeFlag)
+  dvData[['stat1TimeSeriesEst']] <- parseDVTimeSeries(reportObject, 'firstDownChain', 'downChainDescriptions1', timezone, excludeZeroNegativeFlag, estimated=TRUE)
+  dvData[['stat2TimeSeries']] <- parseDVTimeSeries(reportObject, 'secondDownChain', 'downChainDescriptions2', timezone, excludeZeroNegativeFlag)
+  dvData[['stat2TimeSeriesEst']] <- parseDVTimeSeries(reportObject, 'secondDownChain', 'downChainDescriptions2', timezone, excludeZeroNegativeFlag, estimated=TRUE)
+  dvData[['stat3TimeSeries']] <- parseDVTimeSeries(reportObject, 'thirdDownChain', 'downChainDescriptions3', timezone, excludeZeroNegativeFlag)
+  dvData[['stat3TimeSeriesEst']] <- parseDVTimeSeries(reportObject, 'thirdDownChain', 'downChainDescriptions3', timezone, excludeZeroNegativeFlag, estimated=TRUE)
+  dvData[['comparisonTimeSeries']] <- parseDVTimeSeries(reportObject, 'comparisonSeries', 'comparisonSeriesDescriptions', timezone, excludeZeroNegativeFlag)
+  dvData[['comparisonTimeSeriesEst']] <- parseDVTimeSeries(reportObject, 'comparisonSeries', 'comparisonSeriesDescriptions', timezone, excludeZeroNegativeFlag, estimated=TRUE)
+  primaryPresentTS <- names(dvData)[1]
+
+  #Validate Basic Plot Data
+  if(is.null(primaryPresentTS)){
+    return(NULL)
+  }
+
+  #Get Estimated / Non-Estimated Edges
+  if(!is.null(dvData[['stat1TimeSeries']]) && !is.null(dvData[['stat1TimeSeriesEst']])){
+    dvData[['estimated1Edges']] <- getEstimatedEdges(dvData[['stat1TimeSeries']][['points']], dvData[['stat1TimeSeriesEst']][['points']])
+  }
+
+  if(!is.null(dvData[['stat2TimeSeries']]) && !is.null(dvData[['stat2TimeSeriesEst']])){
+    dvData[['estimated2Edges']] <- getEstimatedEdges(dvData[['stat2TimeSeries']][['points']], dvData[['stat2TimeSeriesEst']][['points']])
+  }
+
+  if(!is.null(dvData[['stat3TimeSeries']]) && !is.null(dvData[['stat3TimeSeriesEst']])){
+    dvData[['estimated3Edges']] <- getEstimatedEdges(dvData[['stat3TimeSeries']][['points']], dvData[['stat3TimeSeries']][['points']])
+  }
+
+  if(!is.null(dvData[['comparisonTimeSeries']]) && !is.null(dvData[['comparisonTimeSeriesEst']])){
+    dvData[['comparisonEdges']] <- getEstimatedEdges(dvData[['comparisonTimeSeries']][['points']], dvData[['comparisonTimeSeriesEst']][['points']])
+  }
+
+  #Get Additional Plot Data
+  dvData[['gw_level']] <- parseDVGroundWaterLevels(reportObject)
+  dvData[['meas_Q']] <- parseDVFieldVisitMeasurements(reportObject)
+  dvData <- append(dvData, parseDVMinMaxIVs(reportObject, timezone, dvData[[primaryPresentTS]][['type']], invertedFlag, excludeMinMaxFlag, excludeZeroNegativeFlag))
+  dvData <- append(dvData, parseDVApprovals(dvData[[primaryPresentTS]], timezone))
```

Comment:
I switched to using append because those functions are returning lists of several things instead of a single data frame. Might be cleaner to switch them to returning individual data frames, though that would require changes the the readApprovalBar function in utils-approvals.